examples/private-endpoint-Staticip/main.tf

```diff
@@ -0,0 +1,266 @@
+    random = {
+      source  = "hashicorp/random"
+      version = ">= 3.5.0, < 4.0.0"
+    }
+  }
+}
+provider "azurerm" {
+  features {
+    resource_group {
+      prevent_deletion_if_contains_resources = false
+    }
+  }
+  skip_provider_registration = true
+  storage_use_azuread        = true
+}
+
+locals {
+  test_regions = ["eastus", "eastus2", "westus2", "westus3"]
+}
+# This allows us to randomize the region for the resource group.
+resource "random_integer" "region_index" {
+  max = length(local.test_regions) - 1
+  min = 0
+}
+# This allow use to randomize the name of resources
+resource "random_string" "this" {
+  length  = 6
+  special = false
+  upper   = false
+}
+# This ensures we have unique CAF compliant names for our resources.
+module "naming" {
+  source  = "Azure/naming/azurerm"
+  version = "0.4.0"
+}
+
+
+resource "azurerm_resource_group" "this" {
+  location = local.test_regions[random_integer.region_index.result]
+  name     = module.naming.resource_group.name_unique
+}
+
+resource "azurerm_virtual_network" "vnet" {
+  address_space       = ["192.168.0.0/16"]
+  location            = azurerm_resource_group.this.location
+  name                = module.naming.virtual_network.name_unique
+  resource_group_name = azurerm_resource_group.this.name
+}
+
+resource "azurerm_subnet" "private" {
+  address_prefixes     = ["192.168.0.0/24"]
+  name                 = module.naming.subnet.name_unique
+  resource_group_name  = azurerm_resource_group.this.name
+  virtual_network_name = azurerm_virtual_network.vnet.name
+  service_endpoints    = ["Microsoft.Storage"]
+}
+
+resource "azurerm_network_security_group" "nsg" {
+  location            = azurerm_resource_group.this.location
+  name                = module.naming.network_security_group.name_unique
+  resource_group_name = azurerm_resource_group.this.name
+}
+
+resource "azurerm_subnet_network_security_group_association" "private" {
+  network_security_group_id = azurerm_network_security_group.nsg.id
+  subnet_id                 = azurerm_subnet.private.id
+}
+
+resource "azurerm_network_security_rule" "no_internet" {
+  access                      = "Deny"
+  direction                   = "Outbound"
+  name                        = module.naming.network_security_rule.name_unique
+  network_security_group_name = azurerm_network_security_group.nsg.name
+  priority                    = 100
+  protocol                    = "*"
+  resource_group_name         = azurerm_resource_group.this.name
+  destination_address_prefix  = "Internet"
+  destination_port_range      = "*"
+  source_address_prefix       = azurerm_subnet.private.address_prefixes[0]
+  source_port_range           = "*"
+}
+
+locals {
+  # Provide static ip address for each endpoint. Make sure to use a unique IP address for each endpoint to avoide conflicts.
+  endpoints = {
+    blob  = "192.168.0.7"
+    queue = "192.168.0.8"
+    table = "192.168.0.9"
+    file  = "192.168.0.10"
+  }
+}
+
+module "public_ip" {
```

Comment:
There ss an AVM module for public ip now, would be better to switch to that if using a module.

Reply:
I will make this change with a different PR